TestFoundation/TestNSValue.swift

```diff
@@ -0,0 +1,121 @@
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2015 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See http://swift.org/LICENSE.txt for license information
+// See http://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+
+
+#if DEPLOYMENT_RUNTIME_OBJC || os(Linux)
+    import Foundation
+    import XCTest
+#else
+    import SwiftFoundation
+    import SwiftXCTest
+#endif
+
+
+class TestNSValue : XCTestCase {
+    var allTests : [(String, () -> Void)] {
+        return [
+            ( "test_valueWithLong", test_valueWithLong ),
+            ( "test_valueWithCGPoint", test_valueWithCGPoint ),
+            ( "test_valueWithCGSize", test_valueWithCGSize ),
+            ( "test_valueWithCGRect", test_valueWithCGRect ),
+            ( "test_valueWithNSEdgeInsets", test_valueWithNSEdgeInsets ),
+            ( "test_valueWithNSRange", test_valueWithNSRange ),
+            ( "test_valueWithShortArray", test_valueWithShortArray ),
+            ( "test_valueWithULongLongArray", test_valueWithULongLongArray ),
+            ( "test_valueWithCharPtr", test_valueWithULongLongArray ),
+        ]
+    }
+    
+    func test_valueWithCGPoint() {
+        let point = CGPoint(x: CGFloat(1.0), y: CGFloat(2.0234))
+        let value = NSValue(point: point)
+        XCTAssertEqual(value.pointValue, point)
+        
+        var expected = CGPoint()
+        value.getValue(&expected)
+        XCTAssertEqual(expected, point)
+    }
+    
+    func test_valueWithCGSize() {
+        let size = CGSize(width: CGFloat(1123.234), height: CGFloat(3452.234))
+        let value = NSValue(size: size)
+        XCTAssertEqual(value.sizeValue, size)
+        
+        var expected = CGSize()
+        value.getValue(&expected)
+        XCTAssertEqual(expected, size)
+    }
+    
+    func test_valueWithCGRect() {
+        let point = CGPoint(x: CGFloat(1.0), y: CGFloat(2.0234))
+        let size = CGSize(width: CGFloat(1123.234), height: CGFloat(3452.234))
+        let rect = CGRect(origin: point, size: size)
+        let value = NSValue(rect: rect)
+        XCTAssertEqual(value.rectValue, rect)
+        
+        var expected = CGRect()
+        value.getValue(&expected)
+        XCTAssertEqual(expected, rect)
+    }
+    
+    func test_valueWithNSRange() {
+        let range = NSMakeRange(1, 2)
+        let value = NSValue(range: range)
+        XCTAssertEqual(value.rangeValue, range)
+        
+        var expected = NSRange()
+        value.getValue(&expected)
+        XCTAssertEqual(expected, range)
```

Comment:
Running this against darwin yields this error; so NSRange is not equatable in swift
```
error: cannot invoke 'XCTAssertEqual' with an argument list of type '(NSRange, NSRange)'
        XCTAssertEqual(value.rangeValue, range)
```